src/web/src/services/YacsService.js

```diff
@@ -121,3 +121,32 @@ export const getStudentCourses = (user_info) =>
       params: user_info,
     })
     .then((res) => res.data);
+
+export const getCoursesBySearch = (semester, search) =>
```

Comment:
Instead of defining a separate function, can we just add an optional `search` parameter to `getCourses`? This ties in to using the same route mentioned above